### PR TITLE
Howto: Working With Pipelines

### PR DESCRIPTION
**Pull request by svekars**

### CODE REVIEW

File: doc/how-tos/working-with-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# Working with Pipelines
+
+A typical Pachyderm workflow involves multiple iterations of
+experimenting with your code and pipeline specs.
+
+Before you read this section, make sure that you
+understand basic Pachyderm concepts described in
```
Comment (JoeyZwicker) [inline doc/how-tos/working-with-pipelines.md]:
```suggestion
understand basic Pachyderm pipeline concepts described in
```

File: doc/how-tos/working-with-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+[Concepts](../concepts/index.rst).
```
Comment (JoeyZwicker) [inline doc/how-tos/working-with-pipelines.md]: link to pipeline concepts?

File: doc/how-tos/removing_data_from_pachyderm.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+```bash
+$ pachctl delete commit <repo>@<branch-or-commit-id>
+```
+
+When you delete a bad commit, Pachyderm performs the following actions:
+
+- Deletes the commit metadata.
+- Changes HEADs of all the branches that had the bad commit as their
+  HEAD to the bad commit parent. If the bad commit does not have
```
Comment (JoeyZwicker) [inline doc/how-tos/removing_data_from_pachyderm.md]:
```suggestion
  HEAD to the bad commit's parent. If the bad commit does not have
```

File: doc/how-tos/removing_data_from_pachyderm.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+In Git terms, `pachctl delete commit` is equivalent to squashing a
+commit out of existence, such as with the `git reset --hard` command.
+The `delete commit` command is not equivalent to reverting a
+commit in Git. The reason for this
+behavior is that the semantics of revert can get ambiguous
+when the files that are being reverted have been
+otherwise modified. Because Pachyderm is a centralized system
+and the volume of data that you typically store in Pachyderm is
+large, the concept of merge conflicts that you might run into when
```
Comment (JoeyZwicker) [inline doc/how-tos/removing_data_from_pachyderm.md]: > ...  the concept of merge conflicts that you might run into when
you work with Git is irrelevant.

this sentence in confusing. change to:

...large, merge conflicts can quickly become untenable so Pachyderm prevents merge conflicts entirely.

File: doc/how-tos/removing_data_from_pachyderm.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+those commits might also have the data that you might want to
+keep.
+
+**Warning:** The `pachctl delete file` command is irreversible. The
+files that you delete cannot be restored. If you are not sure
+about deleting a file, store a copy of it outside of Pachyderm.
+
+To delete a file in an older commit, complete the following steps:
+
+1. Start a new commit:
```
Comment (JoeyZwicker) [inline doc/how-tos/removing_data_from_pachyderm.md]: these commands should be a little more fleshed out

File: doc/how-tos/removing_data_from_pachyderm.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+   ```bash
+   $ pachctl start commit
```
Comment (JoeyZwicker) [inline doc/how-tos/removing_data_from_pachyderm.md]:
```suggestion
   $ pachctl start commit <repo>@<branch>
```

File: doc/how-tos/removing_data_from_pachyderm.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+and the volume of data that you typically store in Pachyderm is
+large, the concept of merge conflicts that you might run into when
+you work with Git is irrelevant.
+
+To resolve issues with the commits that are not at the tip of the
+branch, you can try to delete the children commits. However,
+those commits might also have the data that you might want to
+keep.
+
+**Warning:** The `pachctl delete file` command is irreversible. The
```
Comment (JoeyZwicker) [inline doc/how-tos/removing_data_from_pachyderm.md]: This warning doesnt quite make sense to me. delete file is reversible if that file exists in another commit and you can retrieve it there. I'm not really sure why we have this warning here, perhaps just remove it?

File: doc/how-tos/removing_data_from_pachyderm.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+1. Start a new commit:
+
+   ```bash
+   $ pachctl start commit
+   ```
+
+1. Delete all corrupted files from the newly opened commit:
+
+   ```bash
+   $ pachctl delete file
```
Comment (JoeyZwicker) [inline doc/how-tos/removing_data_from_pachyderm.md]:
```suggestion
   $ pachctl delete file <repo>@<branch or commitID>:/bad/files/you/want/removed
```

File: doc/how-tos/removing_data_from_pachyderm.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+   ```
+
+1. Finish the commit:
+
+   ```bash
+   $ pachctl finish commit
```
Comment (JoeyZwicker) [inline doc/how-tos/removing_data_from_pachyderm.md]:
```suggestion
   $ pachctl finish commit <repo>@<branch>
```

File: doc/how-tos/removing_data_from_pachyderm.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+disk.
+
+If you have accidentally committed sensitive data and you need to
+ensure that it is immediately erased and inaccessible, complete the
+following steps:
+
+1. Delete all the references to data as described in
+[Delete Old Commits](delete-old-commits).
+
+1. Run the `garbage-collect` script:
```
Comment (JoeyZwicker) [inline doc/how-tos/removing_data_from_pachyderm.md]:
```suggestion
1. Run `garbage-collect`:
```

File: doc/how-tos/removing_data_from_pachyderm.md (hunk 7 above)
Comment (svekars) [reply doc/how-tos/removing_data_from_pachyderm.md]: removed

**Comment by svekars** [comment]
Yes, it was already reviewed and no new changes added. Not sure what was wrong with the diff.

**Pull request merged**